### PR TITLE
Add kernel image filters

### PR DESCRIPTION
**Pull request by fbudin69500**

### CODE REVIEW

File: Utilities/SimpleITKJSONDream3DFilterCreationTemplates/KernelImageFilterTemplate.cpp [hunk 1]
```diff
@@ -0,0 +1,215 @@
...
+  sitkBall,
+  sitkBox,
+  sitkCross,
+  sitkPolygon3,
+  sitkPolygon4,
+  sitkPolygon5,
+  sitkPolygon6,
+  sitkPolygon7,
+  sitkPolygon8,
+  sitkPolygon9
```
Comment (thewtex) [inline Utilities/SimpleITKJSONDream3DFilterCreationTemplates/KernelImageFilterTemplate.cpp]: Remove Polygon members?

File: Utilities/SimpleITKJSONDream3DFilterCreationTemplates/KernelImageFilterTemplateTest.cpp [hunk 2]
```diff
@@ -0,0 +1,66 @@
...
+  sitkBall,
+  sitkBox,
+  sitkCross,
+  sitkPolygon3,
+  sitkPolygon4,
+  sitkPolygon5,
+  sitkPolygon6,
+  sitkPolygon7,
+  sitkPolygon8,
+  sitkPolygon9
```
Comment (thewtex) [inline Utilities/SimpleITKJSONDream3DFilterCreationTemplates/KernelImageFilterTemplateTest.cpp]: Remove Polygon members?

File: Documentation/ITKImageProcessingFilters/ITKBlackTopHatImage.md [hunk 3]
```diff
@@ -0,0 +1,49 @@
+ITKBlackTopHatImage {#itkblacktophatimage}
+===================
+
+## Group (Subgroup) ##
+ITKImageProcessing (ITKImageProcessing)
+
+## Description ##
+Black top hat extract local minima that are smaller than the structuring element.
+
+Black top hat extract local minima that are smaller than the structuring element. It subtract the background in the input image. The output of the filter transforms the black peaks in white peaks.
```
Comment (thewtex) [inline Documentation/ITKImageProcessingFilters/ITKBlackTopHatImage.md]: subtract -> subtracts

The output of the filter returns black structures that are smaller than the structuring element.

Please also fix these in the ITK documentation.

Comment (fbudin69500) [reply Documentation/ITKImageProcessingFilters/ITKBlackTopHatImage.md]: This should be corrected in ITK :)

File: Utilities/SimpleITKJSONDream3DFilterCreationTemplates/KernelImageFilterTemplate.cpp (hunk 1 above)
Comment (fbudin69500) [reply Utilities/SimpleITKJSONDream3DFilterCreationTemplates/KernelImageFilterTemplate.cpp]: Makes sense.

**Review by thewtex — APPROVED** [review]
:1st_place_medal:

**Comment by fbudin69500** [comment]
@thewtex Addition of the WhiteTopHat and a little debugging later, here is the updated PR.

File: Documentation/ITKImageProcessingFilters/ITKWhiteTopHatImage.md [hunk 4]
```diff
@@ -0,0 +1,47 @@
+ITKWhiteTopHatImage {#itkwhitetophatimage}
+===================
+
+## Group (Subgroup) ##
+ITKImageProcessing (ITKImageProcessing)
+
+## Description ##
+White top hat extract local maxima that are larger than the structuring element.
```
Comment (thewtex) [inline Documentation/ITKImageProcessingFilters/ITKWhiteTopHatImage.md:8]: extract -> extracts

**Review by thewtex — APPROVED** [review]

**Comment by fbudin69500** [comment]
@thewtex : I think I am done working on this branch :)

**Comment by thewtex** [comment]
@fbudin69500 Wow! :-O Excellent!

One tweak -- GrayscaleGeodesicErodeImageFilter and GrayscaleGeodesicDilateImageFilter are deprecated and can be removed. This could be added in an Issue if not done on this branch.

**Comment by imikejackson** [comment]
+1

**Review by thewtex — APPROVED** [review]

**Pull request merged**